webextension/background/auth.js

```diff
@@ -13,18 +13,24 @@ this.auth = (function() {
   let abTests = {};
   let accountId = null;
 
-  const registrationInfoFetched = catcher.watchPromise(browser.storage.local.get(["registrationInfo", "abTests"]).then((result) => {
-    if (result.abTests) {
-      abTests = result.abTests;
-    }
-    if (result.registrationInfo) {
-      registrationInfo = result.registrationInfo;
-    } else {
+  const fetchStoredInfo = catcher.watchPromise(
+    browser.storage.local.get(["registrationInfo", "abTests"]).then((result) => {
+      if (result.abTests) {
+        abTests = result.abTests;
+      }
+      if (result.registrationInfo) {
+        registrationInfo = result.registrationInfo;
+      }
+  }));
+
+  function getRegistrationInfo() {
+    if (!registrationInfo) {
```

Comment:
This is the main change: break up the fetch from local storage and id generation.